### PR TITLE
Annotate a few function arguments in doxygen.

### PR DESCRIPTION
**Pull request by bangerth**
Plus minor other fixes.

### CODE REVIEW

File: include/deal.II/dofs/dof_tools.h [hunk 1]
```diff
@@ -2319,19 +2319,19 @@ namespace DoFTools
    * @note The precondition to this function that the output argument needs to
    * have size equal to the total number of degrees of freedom makes this
    * function unsuitable for the case that the given DoFHandler object derives
-   * from a parallel::distributed::Triangulation object.  Consequently, this
+   * from a parallel::Triangulation object (or any of the classes derived from
```
Comment (peterrum) [inline include/deal.II/dofs/dof_tools.h]:
```suggestion
   * from a parallel::TriangulationBase object (or any of the classes derived from
```

File: include/deal.II/dofs/dof_tools.h [hunk 2]
```diff
@@ -2359,7 +2358,8 @@ namespace DoFTools
    * with one entry for each global degree of freedom, but instead a map that
    * maps from the DoFs index to its location. The point of this function is
    * that it is also usable in cases where the DoFHandler is based on a
-   * parallel::distributed::Triangulation object. In such cases, each
+   * parallel::Triangulation object (or any of the classes derived from
```
Comment (peterrum) [inline include/deal.II/dofs/dof_tools.h]: The same here.

**Comment by masterleinad** [comment]
You will also need to run then indenting script.

**Comment by bangerth** [comment]
Updates as requested.

**Pull request merged**